runtime/vm/ObjectFieldInfo.cpp

```diff
@@ -44,6 +45,23 @@ ObjectFieldInfo::countInstanceFields(void)
 		const U_32 modifiers = field->modifiers;
 		if (J9_ARE_NO_BITS_SET(modifiers, J9AccStatic) ) {
 
+#ifdef J9VM_OPT_VALHALLA_VALUE_TYPES
+			J9UTF8 *fieldSig = J9ROMFIELDSHAPE_SIGNATURE(field);
+			U_8 *fieldSigBytes = J9UTF8_DATA(J9ROMFIELDSHAPE_SIGNATURE(field));
+			if ('Q' == *fieldSigBytes) {
+				J9Class *fieldClass = hashClassTableAt(_classLoader, fieldSigBytes + 1, J9UTF8_LENGTH(fieldSig) - 2);
```

Comment:
No possibility of failure to find?

Reply:
I'd rather see this is a call to internalFindClassUTF8 with flags preventing java code from running (find existing only, no exception throw). Should also assert that the returned value is not NULL.